innkeeper/client.go

```diff
@@ -390,16 +442,34 @@ func (c *Client) LoadAll() ([]*eskip.Route, error) {
 }
 
 // Returns all new and deleted routes from Innkeeper since the last LoadAll request.
-func (c *Client) LoadUpdate() ([]*eskip.Route, []string, error) {
+func (c *Client) LoadUpdate() (eskip.RouteList, []string, error) {
 	d, err := c.requestData(true, c.opts.Address+fmt.Sprintf(updatePathFmt, c.lastChanged))
 	if err != nil {
 		return nil, nil, err
 	}
 
-	routes, deleted, lastChanged := convertData(d, c.preRouteFilters, c.postRouteFilters)
+	routes, deleted, lastChanged := convertJsonToEskip(d, c.preRouteFilters, c.postRouteFilters)
 	if lastChanged > c.lastChanged {
 		c.lastChanged = lastChanged
 	}
 
 	return routes, deleted, nil
 }
+
+func (c *Client) UpsertAll(routes eskip.RouteList) error {
+	// convert the routes to the innkeeper json structs
+	data := convertEskipToInnkeeper(routes)
+
+	for _, route := range data {
+		err := c.writeRoute(c.opts.Address+allRoutesPath, route)
```

Comment:
has the Innkeeper API changed? If it still only does insert and delete, then this function shouldn't be called upsert. Upsert means update if it exists, or insert if it doesn't.